### PR TITLE
fix(init): accept an existing directory unless a scaffold file would be overwritten

### PR DESCRIPTION
**Pull request by kojiwakayama**
## Summary
Follow-up to #3983 and #3985.

- `veryfront init app` refused **any** existing `app/` — including an empty one (`mkdir app && veryfront init app`) or a fresh clone holding only `.git`. Every mainstream scaffolder accepts those.
- A conflict is now *a file the scaffold would write over*, not the directory existing. `createProject` is the single authority for both the named and current-directory paths (same `findExistingPaths` check #3985 introduced); the two directory-existence checks in `initCommand` are removed. The refusal names the files:
  ```
  Directory "app" already contains README.md. Use --force to overwrite.
  ```
- `.gitignore` is merged, never a conflict. Behaviour change to be aware of: the interactive wizard now runs before a refusal for a taken name (the old pre-wizard check could not know the template's file list). The message it ends on says exactly what is in the way.
- Left alone on purpose: the `vf_create_project` MCP tool keeps its own `Directory already exists` check/message (`cli/mcp/tools/catalog-tools.ts`). Aligning it is a separate change.
- `docs/api-reference/veryfront/scaffold.md` pins regenerated with CI's Deno 2.7.7.

Found by the `vf-dx-dogfood` edge-case pass against `veryfront@0.1.1251`; recorded there as friction and deliberately kept out of #3983.

## Test plan
- [x] RED on the stacked base: `createProject` into an empty named dir and beside `.git`/`LICENSE` threw "already exists"; `initCommand` into an empty dir was refused pre-wizard
- [x] GREEN: `cli/commands/init/` (incl. `init.integration.test.ts`, subprocess-level empty-dir + conflict cases), `cli/shared/project-creation.test.ts`, `cli/app/` — 24 files pass
- [x] Pass 2 from a sandbox outside the repo with `deno run -A cli/main.ts`: empty dir → scaffolded; `.git`-only dir → scaffolded beside `.git`; dir with `README.md` → refused, exit 1, file intact; no-name into non-empty cwd → still refused (#3985 guard)
- [x] `deno lint`, `deno fmt --check`, `deno check`, `lint:anti-slop`, `lint:sanitizer-baseline`, `lint:test-typecheck`, `docs:api-reference:check` clean

## Update: refuse a file or a link where the scaffold needs a directory

Accepting an existing target directory lets the scaffold reach states the old
`Directory already exists` check never let it see. One of them wrote outside the
project, so this branch now closes it.

`findExistingPaths` asks whether `app/page.tsx` exists. When `app` is a regular
file that path cannot resolve, so the conflict check reports nothing:

- `app` is a **file**: `README.md` and `AGENTS.md` get written, then
  `ensureDir("app")` fails with a raw `Not a directory (os error 20)`, leaving a
  half scaffold behind.
- `app` is a **link** to another directory: nothing fails at all.
  `veryfront init proj` exits 0, prints `proj ready`, and leaves `page.tsx`,
  `layout.tsx` and `about/page.mdx` in the link target instead of `proj/`.

`createProject` now checks every directory the scaffold has to create, before it
writes anything:

```
Directory "app" already contains app as a file or a link, and the scaffold
needs a directory there. Move it aside or use a different name.
```

- The check runs whatever the conflict policy is. `--force` says you accept your
  own files being replaced, not the scaffold writing somewhere else.
- Every segment is checked, so a real `app/` holding a file at `app/about` is
  caught before `app/page.tsx` is written.
- A real directory that is already there is never blocked, which is the point of
  this PR: `mkdir app && veryfront init app` still scaffolds.
- The project directory itself is not checked. `ln -s /mnt/big/app app &&
  veryfront init app` is a legitimate setup and still works.
- The current-directory path had the same hole before this branch, so
  `cd repo && veryfront init` with a linked `app/` wrote outside the repo on
  `main` too. The check sits in `createProject` and covers both paths.

Nothing is ever overwritten through a link: `findExistingPaths` resolves through
it, so an existing `elsewhere/page.tsx` is still reported as a conflict and
refused with the file byte-identical.

### Verification

- Scenario matrix against the real CLI in a sandbox outside the repo, run on
  `origin/main` and on this branch side by side: target holding `README.md`,
  partial scaffold holding only `app/page.tsx`, symlink to a populated
  directory, a directory where a file is expected, a file where a directory is
  expected, non-empty and unreadable, empty, unrelated files plus an existing
  `.gitignore`, `.git`-only, no-name non-empty cwd, and `--force`. No case where
  `main` refused and this branch overwrites. Where behaviour opens up (empty
  directory, unrelated files, `.git`-only) the user's files are byte-identical
  afterwards and `.gitignore` is merged, not replaced.
- Five new tests for the blocked-directory check, all red without it.
- Mutation: skipping the first segment, ignoring nested segments, treating a
  missing path as blocked, refusing any existing ancestor, refusing only when
  every scaffold path exists, refusing any non-empty directory, dropping
  `deno.json` or the env files from `scaffoldWritePaths`, and removing either
  guard outright are each caught by the suite.
- `deno task typecheck`, `deno task lint:ci`, `deno fmt --check`, and
  `deno test cli/shared/project-creation.test.ts cli/commands/init/` all exit 0.

## Update 2: a link at a scaffold path, and the TUI caller

Codex found two more consequences of accepting an existing directory. Both are fixed.

**A link at the scaffold path itself.** The first preflight only walked the
directories above a path, so a dangling `proj/README.md -> ../outside.md` slipped
through: `findExistingPaths` resolves a dangling link to nothing and reports it
absent. `veryfront init proj` exited 0, printed `proj ready`, and wrote the README
to `outside.md` outside the project. Every segment is now checked, including the
last, and a link anywhere along the path is refused whatever the conflict policy:

```
Directory "proj" already contains README.md as a file or a link the scaffold
cannot write through. Move it aside or use a different name.
```

A real file at a scaffold path is still the ordinary conflict pointing at
`--force`, pinned by a test so this cannot drift into refusing any non-empty
directory. The named target being a link stays allowed on purpose:
`ln -s /mnt/big/app app && veryfront init app` puts the project on another volume
and every file is reachable at the path you named. Only a link you did not name
can surprise you.

**The TUI caller.** `cli/app/operations/project-creation.ts` is the second caller
of `createProject` with a fail policy, and it relied on the directory check this
branch removes. It reserves a new remote slug, scaffolds into `projects/<slug>`,
then writes the link for that slug, so with the check gone it would adopt an
existing `projects/<slug>` holding none of the template files and repoint a
directory that is already another project. It now refuses first. The constraint
lives in that caller rather than in `createProject`, because the two want opposite
things: `veryfront init` accepting an existing directory is the point of this PR,
and a freshly reserved slug wanting a fresh directory is the opposite requirement.

**Considered and not changed:** an orphan `package-lock.json` is rewritten by
`npm install`. A real project is already refused, because `package.json` is in
`scaffoldWritePaths`; `--skip-install` never touches the lockfile; so the exposure
is a lockfile with no manifest, which npm regenerates from the manifest the
scaffold just wrote. Adding lockfiles to the preflight would refuse a directory
holding a stray lockfile, which is exactly the behaviour this PR exists to allow.
Reasoning and evidence are on the thread.

### Verification

- Real CLI, sandbox outside the repo: the dangling-leaf case, the same under
  `--force`, and the same on the no-name path all exit 1 and create nothing,
  inside the project or outside it.
- The three cases this PR opens up still scaffold: empty directory (9 files),
  unrelated files (`notes.txt` intact, `.gitignore` merged), and a `.git`-only
  directory.
- Mutation, 10 mutants, all killed. Both coarse mutants from #3985 (refuse only
  when every scaffold path exists; refuse whenever the directory is non-empty)
  are still killed, and the drift test is still the sole killer of the two
  `scaffoldWritePaths` mutants.
- `deno task typecheck`, `deno task lint:ci`, `deno fmt --check`, and
  `deno test cli/shared/project-creation.test.ts cli/app/ cli/commands/init/`
  (24 files, 334 steps) all exit 0.

### CODE REVIEW

File: cli/shared/project-creation.ts [hunk 1]
```diff
@@ -517,29 +517,22 @@ export async function createProject(
...
-  // is any file the scaffold would write over - a `package.json` with the
-  // author's scripts, a `README.md` - and those are refused the same way.
-  if (projectName === undefined && request.conflictPolicy === "fail") {
+  // A conflict is a file the scaffold would write over - a `package.json` with
+  // the author's scripts, a `README.md` - not the directory existing. So an
+  // empty directory, a fresh clone holding only `.git`, or the working
+  // directory itself (the no-name case) all scaffold, and a `--force` is asked
+  // for only when something would actually be replaced.
+  if (request.conflictPolicy === "fail") {
     const conflicts = await findExistingPaths(projectDir, scaffoldWritePaths(assembly, request));
```
[the remark this answers is not in the document]
Comment (kwakayama) [reply cli/shared/project-creation.ts]: Confirmed on both counts, and fixed.

Reproduced against the real CLI in a sandbox outside the repo, on `c67d0ded7f`:

**A file at an ancestor.** `app/app` is a regular file, `veryfront init app`:

```
exit=1, but AGENTS.md was already written, and the error is
  Not a directory (os error 20): stat '.../s4b/app/app/about'
```

**A link at an ancestor.** `proj/app -> ../elsewhere`, `veryfront init proj`:

```
exit=0
  ✓ proj ready
elsewhere/  page.tsx  layout.tsx  about/page.mdx
proj/       README.md AGENTS.md   package.json tsconfig.json public/favicon.svg
```

The command reports success and three files land outside the project. `origin/main` refuses this case (exit 1) because of the directory-existence check this PR removes, so it is a regression from this branch.

Two things your comment did not cover that the same repro settled:

- The current-directory path already had the hole before this PR. `cd repo && veryfront init` with `repo/app -> ../elsewhere` writes outside the repo on `origin/main` today. The fix sits in `createProject`, so it closes both.
- Nothing is ever *overwritten* through the link. `findExistingPaths` resolves through it, so an existing `elsewhere/page.tsx` is still reported as a conflict and refused with the file intact. The bug is writes landing somewhere you did not name, not data loss.

**Fix:** `createProject` now checks every directory the scaffold has to create, before it writes anything, and refuses when one is already a file or a link:

```
Directory "app" already contains app as a file or a link, and the scaffold
needs a directory there. Move it aside or use a different name.
```

Details worth flagging:

- It runs whatever the conflict policy is. `--force` says you accept your own files being replaced, not the scaffold writing somewhere else entirely.
- Every segment is checked, not just the first, so a real `app/` with a file at `app/about` is caught before `app/page.tsx` is written.
- A real directory that is already there is never blocked, which is the whole point of this PR: `mkdir app && veryfront init app` still scaffolds.
- `lstat` reports a link as "not a directory", so one condition covers both cases. The project directory itself is deliberately not checked: `ln -s /mnt/big/app app && veryfront init app` is a legitimate setup, and it still works.

Verified after the fix, real CLI: the file case, the link case, the link case under `--force`, the nested `app/about` case, and the current-directory case all exit 1 with that message and write nothing, in the project or through the link. An empty directory, a directory holding only `.git`, a directory of unrelated files, and a directory that already has a real `app/` all still scaffold.

Tests in `cli/shared/project-creation.test.ts`, five of them, all red without the check. Mutating the check to skip the first segment, to ignore nested segments, to treat a missing path as blocked, or to refuse any existing ancestor each turns the suite red as well.

**Comment by kwakayama** [comment]
@coderabbitai review

The first review attempt on this PR was rate limited, so the green CodeRabbit check on the earlier head carried no signal. The branch has since gained a commit that closes a path where the scaffold wrote outside the project directory, so a real pass over the current head is worth having.

**Comment by kwakayama** [comment]
@codex review

Your P2 on `cli/shared/project-creation.ts` was correct on both halves and is fixed in `865e20004c`. Worth a look at the new head, since your earlier review was against `c67d0ded7f` and the ancestor preflight you asked for is the main thing that changed.

File: cli/shared/project-creation.ts [hunk 2]
```diff
@@ -504,6 +504,44 @@ async function findExistingPaths(dir: string, paths: string[]): Promise<string[]
...
+  // `lstat` reports a link as "not a directory", which is exactly the answer
+  // this needs. It is optional only for virtual filesystems that have no
+  // links of their own; every runtime this CLI scaffolds on provides it, and
+  // `stat` still catches a plain file in the way if one ever does not.
+  const describe = fs.lstat?.bind(fs) ?? fs.stat.bind(fs);
+  const blocked = new Set<string>();
+
+  for (const path of paths) {
+    const segments = path.split("/").slice(0, -1);
+    for (let depth = 1; depth <= segments.length; depth++) {
```
[the remark this answers is not in the document]
Comment (kwakayama) [reply cli/shared/project-creation.ts:533]: Half right, and the half that is right is a real escape. Fixed.

**The leaf link: confirmed, and worse than the ancestor case I fixed.** Reproduced against `865e20004c` with the real CLI in a sandbox:

```
proj/README.md -> ../outside.md   (dangling)
$ veryfront init proj -t minimal --runtime node --skip-install --skip-env-prompt
exit=0
  ✓ proj ready
$ cat outside.md
# Minimal

A barebones starter with just pages and routing, ...
```

Exit 0, success message, and the README written outside the project. You are right about the mechanism: my loop did `path.split("/").slice(0, -1)`, so it never lstat'd the final segment, and `findExistingPaths` resolves a dangling link to nothing and reports it absent.

`findUnwritablePaths` now walks every segment including the last, and refuses any symlink anywhere along the path, whatever the conflict policy:

```
Directory "proj" already contains README.md as a file or a link the scaffold
cannot write through. Move it aside or use a different name.
```

A real file at a scaffold path is deliberately *not* refused here. That one resolves fine and stays the ordinary conflict pointing at `--force`, which a new test pins so this cannot drift into an over-refusal.

Verified after the fix, real CLI: the dangling-leaf case, the same case under `--force`, and the same case on the no-name current-directory path all exit 1 and create nothing, inside the project or outside it.

**The root target: deliberate, and I am keeping it.** When the named target is itself a directory symlink, `ln -s /mnt/big/projects/app app && veryfront init app`, the scaffold lands in the link target and every file is reachable at the path you named. `cd app` works, all nine files are there. That is a normal way to put a project on another volume, and refusing it would break a legitimate setup to protect against nothing: the user named that link themselves and gets exactly what they asked for. It is also not the failure mode of the other cases, where files end up somewhere you never mentioned.

The distinction the check draws is a link *the user did not name*, anywhere below the target, versus the target itself. Only the first can surprise you.

Mutants, all killed: removing the check, removing the symlink branch, dropping the leaf segment again, checking only the first segment, and refusing any path that exists. The symlink branch is now load-bearing rather than redundant with `!isDirectory`, and mutant G ("ignores links") is killed only by the two new leaf tests.

File: cli/shared/project-creation.ts [hunk 3]
```diff
@@ -517,29 +555,33 @@ export async function createProject(
...
-  // author's scripts, a `README.md` - and those are refused the same way.
-  if (projectName === undefined && request.conflictPolicy === "fail") {
-    const conflicts = await findExistingPaths(projectDir, scaffoldWritePaths(assembly, request));
+  // A conflict is a file the scaffold would write over - a `package.json` with
+  // the author's scripts, a `README.md` - not the directory existing. So an
+  // empty directory, a fresh clone holding only `.git`, or the working
+  // directory itself (the no-name case) all scaffold, and a `--force` is asked
+  // for only when something would actually be replaced.
+  if (request.conflictPolicy === "fail") {
+    const conflicts = await findExistingPaths(projectDir, writePaths);
```
[the remark this answers is not in the document]
Comment (kwakayama) [reply cli/shared/project-creation.ts:594]: Mechanically correct, and I am not changing it. Adding lockfiles to the preflight would be an over-refusal that breaks what this PR exists to do.

I reproduced your case and then bounded it. The lockfile is only ever touched in one situation, and it is not a situation with anything to protect:

**A real Node project is already refused, before `npm install` runs.** `package.json` is in `scaffoldWritePaths`, so a directory holding both files never gets that far:

```
proj/ has package-lock.json + package.json
$ veryfront init proj -t minimal --runtime node
exit=1  Directory "proj" already contains package.json. Use --force to overwrite.
lockfile identical=YES   package.json identical=YES
```

**`--skip-install` never touches it**, since `installProjectDependencies` returns early:

```
proj/ has an orphan package-lock.json
$ veryfront init proj -t minimal --runtime node --skip-install
exit=0   lockfile identical=YES
```

So the exposure is exactly: a `package-lock.json` with **no** `package.json`, with installation enabled. That is not a project, it is a leftover. npm treats a lockfile as derived from a manifest, and by the time it runs the scaffold has written a fresh `package.json`, so regenerating the lock to match is npm doing its job rather than the scaffold destroying your work.

Against that, the cost of your suggestion is concrete. `scaffoldWritePaths` is the list of paths the scaffold *writes*, pinned by a drift test that requires every entry to come back in `createdPaths`; a lockfile is not written by the scaffold and would fail that test. Putting it in a separate refusal list instead would make `veryfront init app` refuse a directory containing a stray lockfile, which is precisely the "scaffolds beside files the template does not write" behaviour this PR is for, and precisely the coarse over-refusal two mutants in this suite exist to catch.

What would change my mind is `npm install` mutating something the scaffold does not own and cannot regenerate. It does not: it rewrites a derived file to match a manifest that now exists.

Marking as considered and rejected on the merits rather than unaddressed. If you want belt and braces here it is a separate change with its own trade-off, not a line in this PR.

[the remark this answers is not in the document]
Comment (kwakayama) [reply cli/shared/project-creation.ts:594]: Correct, and the most valuable of the three. This was a second surface I widened without noticing. Fixed.

Confirmed at the `createProject` level with the exact arguments the TUI passes, including `includePackageMetadata: false`, which makes the collision surface *smaller* and adoption therefore more likely:

```
projects/myslug/ contains .veryfront/project.json -> {"projectId":"OTHER-PROJECT"}
                          notes.txt, src/

RESULT: adopted the existing directory, projectDir = <parent>/myslug
created: AGENTS.md, app/about/page.mdx, app/layout.tsx, app/page.tsx,
         public/favicon.svg, README.md, tsconfig.json, .gitignore
```

`createProject` does not itself touch `.veryfront/project.json`, but `resolveOrCreateProject` runs straight after and writes the link for the slug just reserved, so a directory that is already someone else's project gets repointed at a brand new one. On `main` this could not happen: the coarse directory check refused first.

**Fixed in the caller rather than in `createProject`**, because the two callers genuinely want opposite things. `veryfront init` accepting a directory that already exists is the entire point of this PR. The TUI has just reserved a *new* remote slug, so it wants a *new* directory. Putting the constraint in the shared function would force one of them to be wrong.

`cli/app/operations/project-creation.ts` now checks before scaffolding:

```
projects/my-app already exists. Remove it or choose a different name.
```

Test added to `cli/app/operations/project-creation.test.ts` alongside the existing one, using the same fetch stub: a `projects/my-app` already linked to `proj_someone_else` and holding an unrelated file. It asserts the refusal, that the existing link still reads `proj_someone_else`, that the unrelated file is byte-identical, and that no scaffold file landed in the directory. It fails on `865e20004c` and passes now.

**Comment by kwakayama** [comment]
@codex review

Your three P2s on `865e20004c` are all answered in `090585fd1d`. The leaf-link escape and the TUI adoption regression are fixed with tests; the lockfile one is answered on its thread with the reasoning and the evidence bounding it. Worth a pass over the new head.

**Comment by kojiwakayama** [comment]
## Deep review — merge confidence **55/100** (→ 85+ once the one blocker below lands)

**Verdict:** right approach, real-CLI matrix holds up for everything the body lists — except one reproducible hole in the exact class the last two commits exist to close.

### What I verified (isolated worktree at 090585fd1)
- `cli/shared/project-creation.test.ts` + `cli/app/operations/project-creation.test.ts` + `cli/commands/init/` → 18 files, 193 steps, 0 failed (1m55s). `deno check` on the touched entrypoints exit 0. Merges clean with `origin/main`. Docs diff is pin-churn only.
- Real-CLI sandbox **outside the repo** (`deno run -A cli/main.ts init proj --template minimal --skip-install --skip-env-prompt`), 17 cases. All of these behave as the body says: empty dir → scaffolds; `README.md` present → exit 1 naming README.md, file intact; `app -> ../elsewhere` → refused, nothing written; dangling `README.md -> ../outside.md` → refused; `app/about -> …` → refused; `public -> …` → refused; `proj -> real-empty-dir` → allowed; `--force` with `app` link → still refused; no-name path with linked `app` → refused; `--force` with real README → overwrites.

### BLOCKING
**`.gitignore` is the one scaffold write not covered by the link preflight, and it escapes the project.** `scaffoldWritePaths` (`cli/shared/project-creation.ts:490-501`) deliberately omits `.gitignore` so it merges instead of conflicting — but `findUnwritablePaths` walks only `writePaths`, so `<target>/.gitignore` is never `lstat`'ed, and `writeGitignore` (`:374-386`) writes through whatever is there.

Reproduced twice (reviewer agent, then independently by me) with the real CLI:
```
mkdir proj && ln -s ../outside-gitignore.txt proj/.gitignore
veryfront init proj --template minimal --skip-install --skip-env-prompt
# exit 0, "proj ready" — and ../outside-gitignore.txt now exists OUTSIDE proj/ with the full default gitignore
```
(`readTextFile` on the dangling link fails → `undefined` → default content is written through the link.) A link to an *existing* outside file gets rewritten with merged content; same under `--force`; same on the no-name `cd dir && veryfront init` path.

This contradicts the body's "a link anywhere along the path is refused whatever the conflict policy" / "Nothing is ever overwritten through a link". On `main` the named path was unreachable (any existing `proj/` was refused), so for `init <name>` this is **newly reachable via this PR**; the no-name path pre-dates it (#3985).

Fix is small: pass `[...writePaths, ".gitignore"]` to `findUnwritablePaths` only (keep it out of `findExistingPaths` so merge semantics stay), plus one test mirroring the dangling-`README.md` case.

### Non-blocking
1. **TUI slug reservation leaks on refusal** (`cli/app/operations/project-creation.ts:47-61`): `reserveProjectSlug` POSTs `/projects` (creates a real remote project) *before* the local `projects/<slug>` exists check, so a refusal leaves an orphan remote project. Pre-existing in shape, but this PR added the check and could cheaply `lstat` `projects/<normalizedSlug>` before reserving. Message text confirmed: `projects/<slug> already exists. Remove it or choose a different name.`
2. **Git-init into an existing checkout**: the wizard defaults "Initialize Git?" to Yes (`interactive-wizard.ts:186-190`) and `initializeGitRepo` runs `git init && git add -A && git commit` (`cli/utils/git.ts:26-60`). A wizard user scaffolding into a fresh clone (the `.git`-only case the PR advertises) or a dir with unrelated files gets a silent "Initial commit" on their checked-out branch that also stages their unrelated untracked files. Non-interactive never runs git. Consider skipping git when `.git` already exists, or at least a doc note.
3. `vf_create_project` divergence (`cli/mcp/tools/catalog-tools.ts:400-402`) is documented only in the PR body; a one-line in-code comment would stop the next reader "fixing" it.
4. TOCTOU between `lstat` preflight and write (no O_NOFOLLOW in the fs abstraction). Normal CLI-scaffolder threat model (attacker already has write access to the dir) — acceptable; state it in the `findUnwritablePaths` docblock.
5. Pre-existing, not introduced here: refusals render as `✗ [unknown-error] Unknown/unclassified error` with the real text under `Detail:` (`createConfigError` path); dangling `<name>` symlink → raw stat ENOENT.
6. `cli/app/operations/project-creation.test.ts` assigns `globalThis.fetch` directly instead of `withMockFetch` — matches the file's pre-existing pattern (line 39), nit.

`.gitignore` merge semantics verified: existing content kept verbatim; appended after a blank line: `# Environment files` + missing `.env`/`.env.local`/`.env.*.local` (exact trimmed-line dedupe), then `.veryfront/` unless present. `node_modules/` is not added to an existing file. Identical bytes rewritten if nothing is missing. Wizard-before-refusal is an acceptable trade.

### Not re-verified
"10 mutants all killed"; `deno task lint:ci` (CI green).

_Reviewed with Claude Code; the blocking case was reproduced with the real CLI in a sandbox outside the repo._

**Comment by kojiwakayama** [comment]
## Deep review

**Merge confidence: 38/100**  
**Recommendation: REQUEST CHANGES**  
**Reviewed head:** `090585fd1d0b` against current `origin/main`

### Finding

- **HIGH / BLOCK:** `cli/shared/project-creation.ts:486` excludes `.gitignore` from `scaffoldWritePaths()`, and `cli/shared/project-creation.ts:575` reuses that list as the only symlink/unwritable preflight. `writeGitignore()` still writes `projectDir/.gitignore` at line 384.

  **Failure scenario:** in a newly accepted existing named directory, `proj/.gitignore -> ../outside.gitignore` passes preflight. The init succeeds and writes generated ignore content outside the selected project directory. This was reproduced on the current head. A directory at `.gitignore` also fails only after other scaffold files have been written, leaving a partial scaffold.

  **Fix:** keep `.gitignore` out of overwrite-conflict detection so merging remains supported, but include it in a separate complete write-target preflight. Reject symlink leaves/ancestors and directory-at-file-leaf cases. Add named-init and `--force` regressions for existing and dangling `.gitignore` symlinks.

### Standards

The single-authority move into `createProject` is good, but the new filesystem trust boundary omits a real write target.

### Spec

Partial. Empty/unrelated directories and normal scaffold-path links are handled, but the stated link-anywhere protection is not complete.

### Architecture

`BLOCK`: overwrite conflicts and safe write targets are different sets; conflating them creates an out-of-directory write.

### Verification

- Five focused init/project-creation test files: passed, 119 steps, but do not cover this target
- Direct symlink and directory repros: confirmed
- `deno check`, format, and `git diff --check`: passed
